Modules/Module_Import/NII2BIDS/xASL_bids_PhoenixProtocolAnalyzer.m

```diff
@@ -14,198 +14,256 @@
+				end
+
+				if ~isempty(sourcePar.sWipMemBlockadFree1)
+					bidsPar.LabelingPulseMaximumGradient = sourcePar.sWipMemBlockadFree1;
+				end
+			else
+				warning('Missing important parameters in Phoenix for Siemens VEPCASL sequence');
+			end
+		else
+			% An unknown version detected, skipping to the default
+			warning('An unknown version of Siemens VEPCASL sequence detected');
+		end
 	end
-	
-	% N4_VD13D and N4_VE11C have also slab_thickness_mm parameter defined and it is unclear if this is imaging FOV or labeling slab thickness...
-	if isfield(bidsPar,'SoftwareVersions') &&...
-			(~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB15A'))||...
-			 ~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB17A'))||...
-			 ~isempty(regexpi(bidsPar.SoftwareVersions,'N4_VB19A')))
-		if isfield(sourcePar,'sGroupArraysPSatdGap') && ~isempty(sourcePar.sGroupArraysPSatdGap)
-			bidsPar.LabelingDistance = sourcePar.sGroupArraysPSatdGap;
+
+	%% 4. Reading a default sequence
+	if ~bSequenceIdentified
+		%TODO
+		% Saw this option also in QUIPSSII
+		%if sourcePar.sAslulMode==4
+		%    bidsPar.BolusCutOffFlag = true;
+		%	bidsPar.BolusCutOffTechnique = 'Q2TIPS';
+		%end
+
+		%TODO - does it really work for other sequences?
+		if ~isempty(sourcePar.alTI0) && ~isempty(sourcePar.alTI2)
+			if sourcePar.alTI0~=100000 && sourcePar.alTI2~=7000000
+				bidsPar.ScanType = 'ASL';
+			elseif sourcePar.alTI0==100000 && sourcePar.alTI2==7000000
+				bidsPar.ScanType = 'PseudoM0';
+			end
 		end
-	
-		if isfield(bidsPar,'ArterialSpinLabelingType') && ~isempty(regexpi(bidsPar.ArterialSpinLabelingType,'pasl')) &&...
-				isfield(sourcePar,'sGroupArraysPSatdThickness') && ~isempty(sourcePar.sGroupArraysPSatdThickness)
-			bidsPar.LabelingSlabThickness = sourcePar.sGroupArraysPSatdThickness;
+
+		if isfield(sourcePar,'sProtConsistencyInfotBaselineString') && ~isempty(sourcePar.sProtConsistencyInfotBaselineString)
+			bidsPar.SoftwareVersions = strrep(sourcePar.sProtConsistencyInfotBaselineString,'"','');
 		end
-	end
-	% The function of this parameter is unclear
-	%if isfield(sourcePar,'Slab_thickness_mm') && ~isempty(sourcePar.Slab_thickness_mm)
-	%	bidsPar.LabelingSlabThickness = sourcePar.Slab_thickness_mm;
-	%end
-	
-	if isfield(sourcePar,'alTE0') && ~isempty(sourcePar.alTE0)
-		bidsPar.EchoTime = sourcePar.alTE0 / 1000 / 1000;
-		lastEchoTime = bidsPar.EchoTime;
-		lastEchoTimeCount = 1;
-		while lastEchoTime > 0
-			lastEchoTimeCount = lastEchoTimeCount + 1;
-			echoTimePar = getPhoenixParameters({['alTE[' num2str(lastEchoTimeCount) ']']},parameterList,0);
-			if isempty(echoTimePar{2})
-				lastEchoTime = 0;
-			else
-				lastEchoTime = str2num(echoTimePar{2})/1000/1000;
-				bidsPar.EchoTime(lastEchoTimeCount) = lastEchoTime;
+
+		% N4_VD13D and N4_VE11C have also slab_thickness_mm parameter defined and it is unclear if this is imaging FOV or labeling slab thickness...
+		if isfield(bidsPar,'SoftwareVersions') &&...
```

Comment:
Lots of if statements throughout the whole function, that probably only you understand :) I don't care, but switch-case constructions could be more readable for yourself and others (including comments if helpful)

Reply:
That was the old code, which will be gradually phased out and rewritten. I just moved that part, not edited it.